mailbox/event/event-cassandra/src/main/java/org/apache/james/mailbox/events/CassandraEventDeadLetters.java

```diff
@@ -39,13 +39,15 @@
     }
 
     @Override
-    public Mono<Void> store(Group registeredGroup, Event failDeliveredEvent, InsertionId insertionId) {
+    public Mono<InsertionId> store(Group registeredGroup, Event failDeliveredEvent) {
         Preconditions.checkArgument(registeredGroup != null, REGISTERED_GROUP_CANNOT_BE_NULL);
         Preconditions.checkArgument(failDeliveredEvent != null, FAIL_DELIVERED_EVENT_CANNOT_BE_NULL);
-        Preconditions.checkArgument(insertionId != null, FAIL_DELIVERED_ID_INSERTION_CANNOT_BE_NULL);
 
-        return cassandraEventDeadLettersDAO.store(registeredGroup, failDeliveredEvent, insertionId)
+        InsertionId insertionId = InsertionId.random();
+        cassandraEventDeadLettersDAO.store(registeredGroup, failDeliveredEvent, insertionId)
```

Comment:
It's not working anymore: you should chain the return of `Mono.just(insertionId)` to the `store` Mono otherwise a `block` on the return of this method won't block on the actual saving into Cassandra